### PR TITLE
Properly handle metadce of MINIMAL_RUNTIME exports

### PR DESCRIPTION
**Pull request by kripken**
We handled imports, but not exports. As a result, some things in
the wasm binary were not cleaned up by metadce. I noticed this
when comparing builds with minimal and normal runtime - the JS
was much smaller in minimal, but the binaries were a little bigger
due to stuff like unnecessary dynCalls, things like stackAlloc, etc.

This adds a metadce test for minimal runtime.

You can see the effects of this optimization in this commit, where
I updated test expectations:

https://github.com/emscripten-core/emscripten/commit/9804332f45f2576af747350c98ba67ebcacb9287

In particular in the metadce test it shrinks the wasm from
267 to 85 bytes, matching the normal runtime's behavior.

### CODE REVIEW

File: tools/ports/bzip2.py [hunk 1]
```diff
@@ -7,19 +7,19 @@
...
     return []
 
-  ports.fetch_project('bzip2', 'https://sourceware.org/pub/bzip2/bzip2-1.0.6.tar.gz', 'bzip2-1.0.6', sha512hash=HASH)
+  ports.fetch_project('bzip2', 'https://github.com/emscripten-ports/bzip2/archive/' + VERSION + '.zip', 'bzip2-' + VERSION, sha512hash=HASH)
 
   def create():
     ports.clear_project_build('bzip2')
 
-    source_path = os.path.join(ports.get_dir(), 'bzip2', 'bzip2-1.0.6')
+    source_path = os.path.join(ports.get_dir(), 'bzip2', 'bzip2-' + VERSION)
```
Comment (juj) [inline tools/ports/bzip2.py]: This change looks unrelated - accidental, or do we want to land this in?

File: tools/acorn-optimizer.js [hunk 2]
```diff
@@ -628,15 +637,68 @@ function emitDCEGraph(ast) {
...
+        emptyOut(node);
+      }
     } else if (node.type === 'FunctionDeclaration') {
       defuns.push(node);
       var name = node.id.name;
       nameToGraphName[name] = getGraphName(name, 'defun');
       emptyOut(node); // ignore this in the second pass; we scan defuns separately
+    } else if (node.type === 'FunctionExpression') {
+      // Check if this is the minimal runtime exports function, which looks like
+      //   function(output) { var asm = output.instance.exports;
```
Comment (juj) [inline tools/acorn-optimizer.js]: Hmm, does this require that the exports look like
```
function(output) { var asm = output.instance.exports;
```
?
In some cases the exports look instead like
```
var asm;
function(output) { asm = output.instance.exports;
```
(in particular `#if !(LibraryManager.has('library_exports.js') && (WASM || WASM_BACKEND)) && !EMBIND` is false)

File: tools/ports/bzip2.py (hunk 1 above)
Comment (kripken) [reply tools/ports/bzip2.py]: That's already landed on master, I added it here to verify Ci is green.

File: tools/acorn-optimizer.js (hunk 2 above)
Comment (kripken) [reply tools/acorn-optimizer.js]: Oh, then maybe there's another pattern then that we should match. But sounds like it's only for fastcomp, though? I'd prefer to not do it in that case unless this is especially urgent for some reason.

Comment (juj) [reply tools/acorn-optimizer.js]: The global `var asm;` occurs 1) if using `library_exports.js` (compiling with `-lexports.js`), or 2) if using embind, or 3) if using asm.js in fastcomp. The cases 1) and 2) also happen in wasm backend.
